### ISSUE TITLE
Implement datastore.TransactionStore with Postgres

### ISSUE THREAD

**Issue by robdefeo** [mailchain/mailchain#515]
Is your feature request related to a problem? Please describe.
The indexer needs to store transactions with Mailchain envelopes so they can be read by the receiver 

Describe the solution you'd like
- [ ]  Create in `cmd/internal/datastore/pq`
- [ ]  Implement with Postgres
- [ ] Implement `datastore.TransactionStore` from `cmd/internal/datastore/transaction.go`
- [ ] create unit tests 
- [ ] see `db/migrations` For table structure
- [ ] see `mappings.go` For how to use uint8 string mappings
- [ ] see `syncstore.go` for example implementation

**Comment by Despire** [comment]
@robdefeo I've been working on this issue and come across something that I'm not sure of.
For what is the [hash slice](https://github.com/mailchain/mailchain/blob/master/cmd/internal/datastore/transaction.go#L22) used here? Is it something different from this [hash](https://github.com/mailchain/mailchain/blob/master/cmd/internal/datastore/transaction.go#L15)?

Also, there seems to no table structure in the `db/migrations` so I've created one, would you agree with the following structure?
```sql
CREATE TABLE IF NOT EXISTS transactions(
    -- Primary Key
    protocol        SMALLINT NOT NULL,
    network         SMALLINT NOT NULL,
    -- Values
    tx_from         BYTEA NOT NULL,
    tx_to           BYTEA NOT NULL,
    tx_data         BYTEA NOT NULL,
    tx_block_hash   BYTEA NOT NULL,
    tx_hash         BYTEA NOT NULL,
    tx_value        BYTEA NOT NULL,
    tx_gas_used     BYTEA NOT NULL,
    tx_gas_price    BYTEA NOT NULL,
    PRIMARY KEY(protocol, network)
);
```

**Comment by robdefeo** [comment]
Hi @Despire [hash slice](https://github.com/mailchain/mailchain/blob/master/cmd/internal/datastore/transaction.go#L22) is used to uniquely identify the transaction. It is the same as this [hash](https://github.com/mailchain/mailchain/blob/master/cmd/internal/datastore/transaction.go#L15).
Thanks for creating the table migration which is needed. This means that the `hash` would

```
```sql
CREATE TABLE IF NOT EXISTS transactions(
    -- Primary Key
    protocol        SMALLINT NOT NULL,
    network         SMALLINT NOT NULL,
    hash         BYTEA NOT NULL,
    -- Values
    from         BYTEA NOT NULL,
    to           BYTEA NOT NULL,
    data         BYTEA NOT NULL,
    block_hash   BYTEA NOT NULL,    
    value        BYTEA NOT NULL,
    gas_used     BYTEA NOT NULL,
    gas_price    BYTEA NOT NULL,
    PRIMARY KEY(protocol, network, hash)
);
```
```

**Comment by robdefeo** [comment]
@Despire I accidently edit your comment (I was trying to copy the raw markdown), I tried to put it back to what you wrote originally. Apologies for any confusion that this has caused.

**Comment by robdefeo** [comment]
closed by #520
